src/runtime/methodbinder.cs

```diff
@@ -761,7 +761,24 @@ internal virtual IntPtr Invoke(IntPtr inst, IntPtr args, IntPtr kw, MethodBase i
                 return t;
             }
 
-            return Converter.ToPython(result, mi.ReturnType);
+            return MakeReturnValue(result, mi.ReturnType);
+        }
+
+        /// <summary>
+        /// Takes care of wrapping an object in an interface object
+        /// in case the method returns a value of a interface type.
+        /// </summary>
+        private IntPtr MakeReturnValue(object retVal, Type retType)
+        {
+            if (retVal != null && retType.IsInterface)
+            {
+                var ifaceObj = (InterfaceObject)ClassManager.GetClass(retType);
+                return CLRObject.GetInstHandle(retVal, ifaceObj.pyHandle);
```

Comment:
One thing we lose out on here is that you cannot register custom converters for interfaces. I should probably restore that ability.

Reply:
Yeah, can't you just set `retVal` in this branch to `ifaceObj` and pass it to the `Converter.ToPython` as usual?

Reply:
Not really. Implementations of `IPyObjectEncoder` will expect to get the underlying CLR object, not an `InterfaceObject`.

I should probably move the logic into `Converter.ToPython` instead. The users will have the ability to encode interface instances any way they like if they want to.

Reply:
I moved the logic, and now user defined encoders should once again be respected.